curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/64498542cab69128ab24e4de.md

```diff
@@ -19,7 +19,7 @@ You should add a second capture group to your `rangeRegex`.
 assert.match(code, /const\s+evalFormula\s*=\s*\(\s*x\s*,\s*cells\s*\)\s*=>\s*{\s*const\s+idToText\s*=\s*(\(\s*id\s*\)|id)\s*=>\s*cells\.find\(\s*(\(\s*cell\s*\)|cell)\s*=>\s*(?:cell\.id\s*===\s*id|id\s*===\s*cell\.id)\s*\)\.value\s*;?\s*const\s+rangeRegex\s*=\s*\/\(\[A-J\]\)\(.*\)\/\s*;?/);
 ```
 
-Your second capture group should have a character class.
+Your first capture group should have a character class.
```

Comment:
Thank you for opening this pull request.

First capture group in the `rangeRegex` is `([A-J])`, but that's not what the test checks. Both test and hint refers to the second capture group.

Hints are kind of leading step-by-step to complete answer:

- You should add a second capture group to your `rangeRegex`.
  `const rangeRegex = /([A-J])()/;`
- Your second capture group should have a character class.
  `const rangeRegex = /([A-J])([])/;`
- Your second capture group should have two character classes.
  `const rangeRegex = /([A-J])([][])/;`
- Your first new character class should match the digits `1` through `9`.
  `const rangeRegex = /([A-J])([1-9][])/;`
- Your second new character class should match the digits `0` through `9`.
  `const rangeRegex = /([A-J])([1-9][0-9])/;`
- Your second new character class should be optional.
  `const rangeRegex = /([A-J])([1-9][0-9]?)/;`

As such the text is not really incorrect. Is this making sense?

Reply:
I see. Thank you for the explanation!